openrouteservice/src/main/java/org/heigit/ors/matrix/algorithms/dijkstra/DijkstraManyToMany.java

```diff
@@ -0,0 +1,565 @@
+
+        for (int source = 0; source < treeEntrySize; ++source) {
+            MultiTreeSPEntryItem currEdgeItem = this.currEdge.getItem(source);
+            double entryWeight = currEdgeItem.getWeight();
+
+            if (entryWeight == Double.POSITIVE_INFINITY || !currEdgeItem.isUpdate())
+                continue;
+
+            if (stoppingCriterion.isEntryLargerThanAllTargets(source, entryWeight))
+                continue;
+
+            MultiTreeSPEntryItem msptSubItem = entry.getItem(source);
+            if (!accept(iter, currEdgeItem.getEdge()))
+                continue;
+
+            configureTurnWeighting(iter, currEdgeItem);
+            double edgeWeight = weighting.calcWeight(iter, swap, currEdgeItem.getOriginalEdge());
+            resetTurnWeighting();
+            if (edgeWeight == Double.POSITIVE_INFINITY)
+                continue;
+
+            double tmpWeight = edgeWeight + entryWeight;
+            if (stoppingCriterion.isEntryLargerThanAllTargets(source, tmpWeight))
+                continue;
+
+            if (msptSubItem.getWeight() > tmpWeight) {
+                msptSubItem.setWeight(tmpWeight);
+                msptSubItem.setEdge(iter.getEdge());
+                msptSubItem.setOriginalEdge(EdgeIteratorStateHelper.getOriginalEdge(iter));
+                msptSubItem.setParent(this.currEdge);
+                msptSubItem.setUpdate(true);
+                addToQueue = true;
+            }
+        }
+        return addToQueue;
+    }
+
+    /**
+     *
+     *
+     * _____SEARCH OUTSIDE CORE DOWNWARDS
+     *
+     */
+
+    /**
+     * Explore a single entry with a downwards filter
+     * @param iter the iterator over the entries
+     */
+    private void exploreEntryDownwards(EdgeIterator iter) {
+        currEdge.resetUpdate(true);
+        currEdge.setVisited(true);
+        if (iter == null)
+            return;
+
+        while (iter.next()) {
+            AveragedMultiTreeSPEntry entry = bestWeightMap.get(iter.getAdjNode());
+
+            if (entry == null) {
+                entry = createEmptyEntry(iter);
+                boolean addToQueue = iterateMultiTreeDownwards(currEdge, iter, entry);
+                if(addToQueue) {
+                    bestWeightMap.put(iter.getAdjNode(), entry);
+                    updateEntryInQueue(entry, true);
+                }
+            } else {
+                boolean addToQueue = iterateMultiTreeDownwards(currEdge, iter, entry);
+                if (!entry.isVisited() || addToQueue) {
+                    // This is the case if the node has been assigned a weight in
+                    // the upwards pass (fillEdges). We need to use it in the
+                    // downwards pass to access lower level nodes, though
+                    // the weight does not have to be reset necessarily
+                    updateEntryInQueue(entry, false);
+                }
+            }
+            if(hasTurnWeighting)
+                turnWeighting.setInORS(true);
```

Comment:
Is this supposed to be called here? If yes please consider using `resetTurnWeighting()`, otherwise remove.

Reply:
Another remain of dev work. Thanks. Removed.

Reply:
You're welcome :)